### ISSUE TITLE
Installation via pip broken under Python 3.9

### ISSUE THREAD

**Issue by bwohlberg** [cabouman/svmbir#224]
Within a Python 3.9 `conda` environment, installation from PyPI via `pip` seems to result in a broken `svmbir` package:
```
$ ipython
Python 3.9.9 | packaged by conda-forge | (main, Dec 20 2021, 02:41:03) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.29.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import svmbir
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-638dc9a72f2c> in <module>
----> 1 import svmbir

/opt/conda/miniconda3/envs/py39/lib/python3.9/site-packages/svmbir/__init__.py in <module>
      1 __version__ = '0.2.6'
----> 2 from .svmbir import *
      3 from .phantom import *
      4 __all__ = ['_svmbir_lib_path','_clear_cache','sino_sort','auto_sigma_x','auto_sigma_p','auto_sigma_y', 'calc_weights','project','backproject','recon']

/opt/conda/miniconda3/envs/py39/lib/python3.9/site-packages/svmbir/svmbir.py in <module>
     12     import svmbir.interface_py_c as ci
     13 else:
---> 14     import svmbir.interface_cy_c as ci
     15 
     16 __svmbir_lib_path = os.path.join(os.path.expanduser('~'), '.cache', 'svmbir')

svmbir/interface_cy_c.pyx in init svmbir.interface_cy_c()

ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```
The `svmbir` binary was built using `gcc` 9.3.0 and `cython` 0.29.26.

It's worth noting that the `svmbir` installation seems to not exhibit this problem if it's installed from source using `CC=gcc pip install  .` as recommended.

I'm assuming that the issue is related to the use of Python 3.9 since we have not previously encountered this problem with Python 3.8, but it is possible that it's really due to some differences in package versions between the standard `conda` packages for 3.8 and 3.9.

**Comment by sjkisner** [comment]
I haven't reproduced the 3.9 error yet on mac or linux.   

Does it happen if you start from a clean environment? I did this,
```
conda create python=3.9 --name test
conda activate test
pip install svmbir
```

It downloads the source, installs dependencies, etc.

I do recognize this ValueError though and it may have been when the installed dependencies' versions didn't match the requirements.txt.

**Comment by sjkisner** [comment]
OK, I can reproduce the error if the installed numpy version is < 1.22.
Immediate fix is to change the requirements to numpy>=1.22

However, it seems the latest numpy version available from conda is only 1.21.2.

**Comment by sjkisner** [comment]
This problem doesn't appear to be specific to python 3.9. I can also reproduce it in 3.8.

As Brendt found, it looks like the error occurs when the numpy version used for the svmbir install is different from the version used when running svmbir.  A solution would be to specify a consistent numpy version for the install (in pyproject.toml) and in the package requirements (in setup.py).  Specifying as `numpy~=1.22.1` tests out OK for me.

With this setting, if an older numpy version is already installed in the environment, the svmbir install should upgrade numpy to 1.22.1.  This works for me when installing with `pip install .` and I believe it should work the same if installing from pypi.

**Comment by sjkisner** [comment]
Fix for this specific error in PR #228. 

Initial tests look like svmbir works for python 3.9 and 3.10.

**Comment by sjkisner** [comment]
Let me summarize this issue. The C API in numpy can change across the minor versions. Therefore it's possible that if the numpy version used to build the svmbir C extension (usually at install time) is different from the numpy version used when running svmbir, an error like at the top of this thread can occur.

The current solution is to pin numpy==1.21.* both in the build (in pyproject.toml) and in the svmbir requirements setup.py.  Seems to work but the consequences,

- Forces a particular numpy on a user environment, obviously
- Support for svmbir is limited to Python 3.7, 3.8, 3.9
- If we pin to numpy==1.22.* we'll be able to support 3.8, 3.9, 3.10 (conda packages for numpy 1.22 are limited)

**Comment by sjkisner** [comment]
This is resolved in v0.2.7
